internal/exec/stages/disks/luks.go

```diff
@@ -306,24 +330,52 @@ func (s *stage) createLuks(config types.Config) error {
 				}
 			}
 
+			// additionally if there are tang servers with advertisement, lets see if they are reachable, and check to make sure the keys match
+			for _, tang := range tangServersWithAdvertisement {
+				u, err := url.Parse(tang.URL)
+				if err != nil {
+					return fmt.Errorf("parsing tang URL: %v", err)
+				}
+				u.Path = path.Join(u.Path, "adv")
+				data, err := s.Fetcher.FetchToBuffer(*u, resource.FetchOptions{})
```

Comment:
The copypasting and the list-building above isn't needed.  We can just have conditionals for with/without adv inside the loop.

Reply:
Good point! I did end up going with a single list, as I felt it was a solid way to count the number of tang servers that I could reach? Does that make sense or am I still doing unnecessary work?

Reply:
>..> true that would have worked as well, Actually still in need of a counter, as we need to know the number of tang servers that have no advertisement.